tests/integration/bigquery/test_filter_and_label.py

```diff
@@ -0,0 +1,26 @@
+"""Tests for the filter_and_label_by_condition function with BigQuery integration."""
+
+from pyretailscience.utils.filter_and_label import filter_and_label_by_condition
+
+SPEND_VALUE = 100
+
+
+def test_filter_and_label_by_condition_with_bigquery(transactions_table):
+    """Test filter_and_label_by_condition with data fetched from BigQuery.
+
+    This test verifies that filter_and_label_by_condition can process data directly from
+    a BigQuery connection using Ibis without throwing exceptions.
+    """
+    limited_table = transactions_table.limit(1000)
+    condition_params = {
+        "high_spend": lambda table: table["unit_spend"] > SPEND_VALUE,
+        "low_spend": lambda table: table["unit_spend"] <= SPEND_VALUE,
+    }
+    conditions = {label: condition_func(limited_table) for label, condition_func in condition_params.items()}
+
+    result = filter_and_label_by_condition(limited_table, conditions)
```

Comment:
This is over complicated. This should work and is simpler.
```suggestion
    conditions = {
        "high_spend": limited_table["unit_spend"] > 100,
        "low_spend": limited_table["unit_spend"] <= 100,
    }
    result = filter_and_label_by_condition(limited_table, conditions)
```